src/main/java/com/dataguise/saas/service/SchedulerService.java

```diff
@@ -0,0 +1,434 @@
+        linkedTaskDetails.setTaskType(scheduleInfoBean.getLinkedTaskDetails().getTaskType().equalsIgnoreCase("DA")
+                ? taskBuilderService.DETECTION_TYPE : taskBuilderService.PROTECTION_TYPE);
+        linkedTaskDetails.setConnectionNames(scheduleInfoBean.getLinkedTaskDetails().getConnectionNames());
+        linkedTaskDetails.setTaskName(scheduleInfoBean.getTaskName());
+        linkedTaskDetails.setTaskId(scheduleInfoBean.getTaskId());
+        return linkedTaskDetails;
+    }
+
+    /**
+     * Delete a scheduled task
+     */
+    public Boolean deleteSchedule(Integer scheduleId) throws BadRequestException, BadGatewayException {
+        if (scheduleId == null) {
+            throw new BadRequestException("Schedule " + ErrorConstants.ERROR_MISSING_PATH_VARIABLE.getErrorMessage());
+        }
+        try {
+            return restDriver.deleteScheduleTask(scheduleId);
+        } catch (DgDrvException e) {
+            throw new BadGatewayException(e.getMessage(), HttpStatus.BAD_GATEWAY);
+        }
+    }
+
+    /**
+     * Update/Edit a schedule
+     *
+     * @param schedulerDto
+     * @param scheduleId
+     */
+    public Boolean editSchedule(@PathVariable Integer scheduleId, @RequestBody SchedulerDTO schedulerDto)
+            throws BadGatewayException, InternalServerException, BadRequestException {
+        Boolean status = false;
+
+        if (schedulerDto.getScheduleName() == null || schedulerDto.getScheduleName().isEmpty()) {
+            logger.error("Schedule name is missing.");
+            throw new BadRequestException(ErrorConstants.ERROR_VALIDATING_SCHEDULER_NAME.getErrorMessage(), HttpStatus.BAD_REQUEST);
+        }
+
+        if (schedulerDto.getScanType() == null) {
+            logger.error("Scan Type is a required parameter and possible values are Weekly/Daily");
+            throw new BadRequestException(ErrorConstants.ERROR_VALIDATING_SCAN_TYPE.getErrorMessage(), HttpStatus.BAD_REQUEST);
+        }
+        if (schedulerDto.getStartDateTime() == null) {
+            logger.error("Start date and time cannot be empty.");
+            throw new BadRequestException(ErrorConstants.ERROR_VALIDATING_START_DATE.getErrorMessage(), HttpStatus.BAD_REQUEST);
+        }
+
+        if (scheduleId != null) {
+            List<SchedulerDTO> existingScheduleList = getSchedule(Optional.of(scheduleId));
+            DgTaskScheduleInfoBean existingSchedule;
+            if (!existingScheduleList.isEmpty()) {
+                existingSchedule = existingScheduleList.get(0).getSchedule();
+            } else {
+                throw new BadGatewayException(ErrorConstants.ERROR_GETTING_SCHEDULER_BY_ID.getErrorMessage(), HttpStatus.BAD_GATEWAY);
+            }
+            logger.info("Existing task info bean for schedule:" + existingSchedule);
+            formatScheduleDTO(schedulerDto);
+            logger.info("Formatted schedule DTO from UI: " + schedulerDto);
```

Comment:
**Reporter**: PMD    
    **Rule**: GuardLogStatement
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/service/SchedulerService.java L283
    
    Logger calls should be surrounded by log level guards.&#10;&#10;&#10;Best Practices https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#guardlogstatement
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-670011890>*